src/node.cc

```diff
@@ -2261,6 +2261,56 @@ static void Uptime(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+static void GetResourceUsage(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  uv_rusage_t rusage;
+  int err = uv_getrusage(&rusage);
+  if (err)
+    return env->ThrowUVException(err, "uv_getrusage");
+
+  Local<Object> resource_usage = Object::New(env->isolate());
```

Comment:
I just did a little benchmarking using 3 possible implementations (after `1e6` iterations):

```js
resourceUsage(): 525.068ms
resourceUsage2(): 418.420ms
resourceUsage3(): 152.699ms
```

`resourceUsage()` is the implementation currently in this PR.

`resourceUsage2()` is using my previously described method: passing an object literal with predefined properties to C++ for setting of values.

`resourceUsage3()` is an implementation that accepts an object generator function defined in JS that is called by C++ and passes all (18) values which are set in an object literal.